### PR TITLE
Feature: travis deploy

### PR DESCRIPTION
**Pull request by jamesjwarren**
This PR adds travis deployment. Releases are triggered on pushing a tag.

Issues to resolve:
- version number is not taken from release tag, it is simply bumped so could get out of sync

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -295,6 +306,11 @@ module.exports = function (grunt) {
         "yuidoc"
     ]);
 
+    grunt.registerTask("release", [
+        "build",
+        "bump"
+    ]);
+
```
Comment (edoparearyee) [inline Gruntfile.js:314]: you have 2 `release` tasks in this gruntfile, i think this one would get overwritten

File: package.json [hunk 2]
```diff
@@ -13,6 +13,7 @@
     "connect-livereload": "~0.5.0",
     "connect-modrewrite": "~0.7.9",
     "coveralls": "~2.11.2",
+    "grunt-bump": "^0.1.0",
     "grunt-contrib-clean": "~0.6.0",
```
Comment (edoparearyee) [inline package.json:17]: you should use `~` instead of `^` to avoid grunt updating the package to a major release which could have breaking functionality

**Comment by edoparearyee** [comment]
@jamesjwarren if grunt bump only increments the version number, we need to find a solution that allows us to specify version number ideally from the git tag

**Comment by jamesjwarren** [comment]
grunt bump accepts an absolute version passed to it so we just need to find a way to extract this info from the git tag.

**Comment by edoparearyee** [comment]
can you pass the tag in the form of a grunt option? something like:
`grunt release -v 1.2.3`

**Comment by edoparearyee** [comment]
This looks good to me, well done! :+1:

**Pull request merged**